src/ControlSystem/ControlErrors/Skew.hpp

```diff
@@ -0,0 +1,277 @@
+        [](const gsl::not_null<DataVector*> inclination_angle,
+           const gsl::not_null<std::array<double, 3>*> intersection_coord,
+           const gsl::not_null<std::array<double, 3>*>
+               intersection_normal_one_form,
+           const ylm::Spherepack& ylm,
+           const ylm::Spherepack::InterpolationInfo<double>& interpolation_info,
+           const auto& normal_one_form, const auto& cartesian_coords) {
+          for (size_t i = 0; i < 3; i++) {
+            ylm.interpolate(make_not_null(&gsl::at(*intersection_coord, i)),
+                            make_not_null(cartesian_coords.get(i).data()),
+                            interpolation_info);
+            ylm.interpolate(
+                make_not_null(&gsl::at(*intersection_normal_one_form, i)),
+                make_not_null(normal_one_form.get(i).data()),
+                interpolation_info);
+          }
+
+          for (size_t i = 0; i < 2; ++i) {
+            (*inclination_angle)[i] =
+                atan2(gsl::at(*intersection_normal_one_form, i + 1),
+                      (*intersection_normal_one_form)[0]);
+            if ((*inclination_angle)[i] < -0.5 * M_PI) {
+              (*inclination_angle)[i] += M_PI;
+            } else if ((*inclination_angle)[i] > +0.5 * M_PI) {
+              (*inclination_angle)[i] -= M_PI;
+            }
+          }
+        };
+
+    // These are the values we need at the theta/phi described above
+    DataVector inclination_angle_a{2, 0.0};
+    DataVector inclination_angle_b{2, 0.0};
+    std::array<double, 3> intersection_coord_a{};
+    std::array<double, 3> intersection_coord_b{};
+    std::array<double, 3> intersection_normal_one_form_a{};
+    std::array<double, 3> intersection_normal_one_form_b{};
+
+    set_intersection(make_not_null(&inclination_angle_a),
+                     make_not_null(&intersection_coord_a),
+                     make_not_null(&intersection_normal_one_form_a), ylm_a,
+                     interpolation_info_a, normal_one_form_a,
+                     cartesian_coords_a);
+    set_intersection(make_not_null(&inclination_angle_b),
+                     make_not_null(&intersection_coord_b),
+                     make_not_null(&intersection_normal_one_form_b), ylm_b,
+                     interpolation_info_b, normal_one_form_b,
+                     cartesian_coords_b);
+
+    const double relative_delta_x =
+        (intersection_coord_a[0] - intersection_coord_b[0]) /
+        (center_a[0] - center_b[0]);
+    // Hardcoded function used in SpEC
+    const double temporal_transition_function =
+        0.5 * (1.0 - tanh(10.0 * relative_delta_x - 5.0));
+
+    // Hardcoded value used in SpEC
+    constexpr double activation_threshold = 0.0025;
+
+    const auto& functions_of_time =
+        Parallel::get<domain::Tags::FunctionsOfTime>(cache);
+    const DataVector func =
+        functions_of_time.at(function_of_time_name)->func(time)[0];
+
+    DataVector control_error = -func;
```

Comment:
I can't tell whether you care about memory allocations in this function or not.  I assumed that's why you're caching things between calls, but then you allocate three extra size-2 DataVectors.